Src/Particle/AMReX_MakeParticle.H

```diff
@@ -0,0 +1,40 @@
+#ifndef AMREX_MAKEPARTICLE_H_
+#define AMREX_MAKEPARTICLE_H_
+
+#include <type_traits>
+
+template< class T >
+struct is_soa_particle
+     : std::integral_constant<
+         bool,
+         T::is_soa_particle
+     > {};
+
+
+template <typename T_ParticleType, class Enable = void>
+struct make_particle
+{
+    template <typename PTD>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    auto&
+    operator() (PTD ptd, int i)
+    {
+        // legacy Particle (AoS)
+        return ptd.m_aos[i];
+    }
+};
+
+template <typename T_ParticleType>
+struct make_particle<T_ParticleType, typename std::enable_if<is_soa_particle<T_ParticleType>::value>::type>
```

Comment:
We are wondering if we can make this as compile-time read-only, to avoid bugs where one writes to this copy of the particle with the new layout - but then the change does not get reflected in the PTD.

Reply:
Oh, I think we can just return a `auto const` ...!

Reply:
Another approach you can consider is `void foo () && = delete` to delete the function for rvalue of the return type.